### DISCUSSION TITLE
Is there a need to restore the dataloader condition

### DISCUSSION THREAD

**Discussion by ryanxingql** [open-mmlab/mmediting#1192] [General]
Hi,
I am wondering that is there a need to restore the dataloader condition.
Every time we restore the model, we only restore the model parameters, optimizer, and scheduler.
The dataloader will start from the beginning, resulting in repeated training over the data in front.
To actually restore the training condition, we may need to restore the dataloader condition, i.e., skip the trained iters or epochs of data.
If this function is necessary, I may try to implement it.

**Comment by zengyh1900** [comment]
Hi @ryanxingql , that's an interesting question!
In my understanding, the data batch from the dataloader is typically shuffled. That means when we resume training, the data batch will be randomly selected from the whole dataset, rather than 'the front'.

**Reply by ryanxingql** [reply]
Thank you for your response. ^ ^

**Reply by ryanxingql** [reply]
If we set a seed for the data loader (i.e., for the data sampler), the data batch will not be randomly shuffled. You understanding may be correct when no seed is set.

**Reply by zengyh1900** [reply]
> If we set a seed for the data loader (i.e., for the data sampler), the data batch will not be randomly shuffled. You understanding may be correct when no seed is set.

You're right!

**Comment by zengyh1900** [comment]
@ryanxingql We appreciate your passion for contributing MMEditing! 
Thanks a lot :star_struck: ! 

We found some of your projects are really interesting!
https://github.com/ryanxingql/powervqe
https://github.com/ryanxingql/winner-ntire22-vqe
https://github.com/ryanxingql/stdf-pytorch

And I have read your paper! https://github.com/ryanxingql/rbqe That's really an interesting idea!

Would you like to contribute these works to MMEditing?
Please don't hesitate! We're here to help if you need it!

**Reply by ryanxingql** [reply]
Thank you so much for your kind words ^ ^ I will contribute to this repo in my spare time.
